### ISSUE TITLE
HTML component preview stopped working after update

### ISSUE THREAD

**Issue by jkphl** [frctl/fractal#211]
For some reasons, the HTML preview of our components stopped working after we updated to the latest releases. It's unclear which exact package or release really introduced the problem, it must've been [Fractal v1.1.0-beta.5](https://github.com/frctl/fractal/releases/tag/v1.1.0-beta.5)+ or [Mandelbrot v1.0.8](https://github.com/frctl/mandelbrot/releases/tag/v1.0.8)+.

Our components themselves still render fine, it's just the HTML preview that doesn't work anymore. There's no expressive error message in the HTML source except `<-- Error rendering component: -->` which comes from the `render.nunj` template in Mandelbrot. Unfortunately, I don't have any experience with Nunjucks etc. so it's really hard for me to find an entry point for debugging.

![fractal-html-preview](https://cloud.githubusercontent.com/assets/4446936/23273379/14e0bc08-f9ff-11e6-93ab-b2ef41fa2065.png)

We're using a custom template engine / rendering adapter (the experimental / still undocumented [fractal-typo3](https://github.com/tollwerk/fractal-typo3)) and none of the "official" ones. I tried @allmarkedup's [`check-errors` suggestion](https://github.com/frctl/fractal/issues/189), but all our components succeed without problems.

Any help highly appreciated. Thanks!

**Comment by allmarkedup** [comment]
@jkphl hmm that is very odd. Can you confirm for me what versions of Fractal and Mandelbrot you are now using? Are they the latest versions?

**Comment by jkphl** [comment]
@allmarkedup Sure:

* Node: 7.0.0
* NPM: 3.10.8
* Fractal: 1.1.3
* Mandelbrot: 1.1.0

I don't expect you to telepathically debug my setup but any pointer where I could start debugging myself would help a lot. Thanks! ;)

**Comment by allmarkedup** [comment]
hey @jkphl really sorry for the slow reply on this... i can't replicate it on my setup here unfortunately. It's a bit of a mystery to me why it's rendering the component correctly, but failing when it outputs the HTML.

It _may_ be helpful to run it in debug mode to see if you get any helpful output to the CLI: `NODE_ENV=debug fractal start` although the debugging info is pretty limited at the moment.

Is it all components that you are seeing this problem with? Or only a subset of components?

**Comment by jkphl** [comment]
@allmarkedup No worries, there's no harm done and we keep on working w/o the HTML view anyway. Yes, we experience this with all components, not just a subset. And yes, it's strange that rendering the components works but the HTML preview doesn't. I'll try switching on the debug mode and let you know. Thanks for your efforts!

**Comment by jkphl** [comment]
Turned out this was due to a synchronous rendering step involved. After consequently switching to Promises everything's running smoothly again. :)

**Comment by nategreen** [comment]
@jkphl Hey, could you elaborate on what the root issue ended up being, and how you solved it? I'm seeing this same thing right now, and it seems like it's only whenever I use the `{{render}}` helper.

**Comment by jkphl** [comment]
@nategreen This was an issue very specific to my setup. We're using a custom rendering engine (our [Fractal-TYPO3 bridge](https://github.com/tollwerk/fractal-typo3)) and when I wrote that initially I used synchronous rendering calls to TYPO3, which were troublesome somehow. Once I fixed that and rewrote the calls to be asynchronous, everything worked like a charm again. I don't recall this being an issue with `{{render}}`, I'm sorry.
